src/component/modal/AddressModal.jsx

```diff
@@ -0,0 +1,125 @@
+import "./addressModal.css";
+import React, { useState } from "react";
+import { useAddress } from "context/address-context";
+import { useAuth } from "context/auth-context";
+
+export const AddressModal = () => {
+  const { isAddModalOpen, setAddModalOpen, setNewAddress } = useAddress();
+  const initState = {
+    firstname: "",
+    lastname: "",
+    street: "",
+    city: "",
+    state: "",
+    pincode: "",
+    phone: "",
+    country: "India",
+  };
+  const [address, setAddress] = useState(initState);
+  const [errormsg, setErrorMsg] = useState("");
+  const changeHandler = (e) => {
```

Comment:
Appreciation - Great implementation of using only one change handler function to update all the states

Reply:
thank you